### PR TITLE
Add connectivity mode option to Arc controller create

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Depends on https://github.com/microsoft/azuredatastudio/pull/13464

Moved Azure fields into separate page with connectivity. Following up with design on this so it may change but good enough for testing for now.

I currently have it showing the connectivity mode in the summary - I would like to follow up with adding the other fields but I want those to be conditional based on whether they're applicable or not so will do that in a separate PR.

Connected Mode selected : 

image

Indirect mode selected : 

image

Summary : 

image

### CODE REVIEW

File: extensions/arc/package.json [hunk 1]
```diff
@@ -212,9 +212,99 @@
...
+                            "defaultValue": "%arc.data.controller.indirect%",
+                            "optionsType": "radio"
+                          }
+                        },
+                        {
+                          "label": "%arc.data.controller.spnclient%",
+                          "variableName": "AZDATA_NB_VAR_SPN_CLIENT_ID",
+                          "type": "text",
+                          "required": true,
+                          "defaultValue": "",
```
Comment (ranasaria) [inline extensions/arc/package.json:273]: for fields with no deafultValue, we should consider if there is a relevant 'placeholder' that we can place to inform the user on how to fill the value.

Comment (Charles-Gagnon) [reply extensions/arc/package.json:273]: I'll be following up with an info link and/or descriptions for these fields once I have something for those - this is to get the feature out for testing initially and then I'll do polishing like that next.

**Comment by ranasaria** [comment]
Are you plannign to add conectivity mode to the summary page in a separate checkin?

**Comment by Charles-Gagnon** [comment]
@ranasaria Read the description 😄 

(Yes)

**Comment by ranasaria** [comment]
Looking at the pictures, when the mode is indirect, I am wondering if we should have more visual cues that the three SPN fields are not relevant. What we have is probably ok for a crawl. But I yearn for some more polishing here later. At a minimum we could consider greying out the corresponding labels as well. Indenting the 3 fields (not dynamically) could be another idea. Filling out a placeHolder that says 'disabled' when those fields are 'disabled' could also be helpful.

**Comment by Charles-Gagnon** [comment]
@ranasaria - Yes, I mentioned this in the other PR. I will be following up with that as well.

**Review by ranasaria — APPROVED** [review]

**Pull request merged**